accord-core/src/main/java/accord/utils/CheckpointIntervalArray.java

```diff
@@ -213,9 +213,6 @@ else if ((ri & BIT29) != 0)
                 forEachScanOrCheckpoint.accept(p1, p2, p3, p4, i);
         }
 
-        if (start == end)
```

Comment:
we spoke in slack, just putting it here.  The disk based version needed this for a reason I can't remember, so that would need to be updated to handle this case.

Reply:
How can I follow up on this? Is there a test that should be failing or something I should look at?

Reply:
I don't remember but here are a few places you should run tests multiple times to see if anything breaks

`org.apache.cassandra.index.accord.CheckpointIntervalArrayIndexTest`
`org.apache.cassandra.utils.RangeTreeTest`
`accord.utils.SearchableRangeListTest`

Reply:
I double checked and those tests touch this function.

Reply:
```
java.lang.IndexOutOfBoundsException: Start is from (0, 10]; attempted to access 10
  at org.apache.cassandra.index.accord.CheckpointIntervalArrayIndex$SortedListReader.fileOffsetStart(CheckpointIntervalArrayIndex.java:336)
  at org.apache.cassandra.index.accord.CheckpointIntervalArrayIndex$SegmentSearcher.lambda$intersects$1(CheckpointIntervalArrayIndex.java:618)
  at accord.utils.CheckpointIntervalArray.forEach(CheckpointIntervalArray.java:219)
  at accord.utils.CheckpointIntervalArray.forEach(CheckpointIntervalArray.java:127)
  at org.apache.cassandra.index.accord.CheckpointIntervalArrayIndex$SegmentSearcher.intersects(CheckpointIntervalArrayIndex.java:612)
  at org.apache.cassandra.index.accord.CheckpointIntervalArrayIndexTest$1.intersects(CheckpointIntervalArrayIndexTest.java:410)
  at org.apache.cassandra.index.accord.CheckpointIntervalArrayIndexTest.fuzz(CheckpointIntervalArrayIndexTest.java:146)
  at org.apache.cassandra.index.accord.CheckpointIntervalArrayIndexTest.lambda$fuzzSmall$1(CheckpointIntervalArrayIndexTest.java:121)
  at accord.utils.Property$SingleBuilder.checkInternal(Property.java:245)
  at accord.utils.Property$SingleBuilder.lambda$check$0(Property.java:230)
  at accord.utils.async.TimeoutUtils.lambda$runBlocking$0(TimeoutUtils.java:39)
  at java.base/java.lang.Thread.run(Thread.java:829)

```

Reply:
looks like a simple fix in `org.apache.cassandra.index.accord.CheckpointIntervalArrayIndex.SegmentSearcher#intersects`

```
searcher.forEach(start, end, (i1, i2, i3, i4, index) -> {
                    stats.matches++;
                    callback.accept(reader.copyTo(accessor.get(indexInput, index), buffer));
                }, (i1, i2, i3, i4, startIdx, endIdx) -> {
                    if (startIdx == endIdx)
                        return;
```